url933dd5727b_AppleLiang_Testing_Project_tgz-pJ8-org_joda_time_chrono_gj_MainTestJ8/src/org/joda/time/TestDateTimeZoneCutover.java

```diff
@@ -0,0 +1,1340 @@
+/*
+ *  Copyright 2001-2013 Stephen Colebourne
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.joda.time;
+
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+import org.joda.time.chrono.GregorianChronology;
+import org.joda.time.tz.DateTimeZoneBuilder;
+
+/**
+ * This class is a JUnit test for DateTimeZone.
+ *
+ * @author Stephen Colebourne
+ */
+public class TestDateTimeZoneCutover extends TestCase {
+
+    public static void main(String[] args) {
+        junit.textui.TestRunner.run(suite());
+    }
+
+    public static TestSuite suite() {
+        return new TestSuite(TestDateTimeZoneCutover.class);
+    }
+
+    public TestDateTimeZoneCutover(String name) {
+        super(name);
+    }
+
+    protected void setUp() throws Exception {
+    }
+
+    protected void tearDown() throws Exception {
+    }
+
+    //-----------------------------------------------------------------------
+    //------------------------ Bug [1710316] --------------------------------
+    //-----------------------------------------------------------------------
+    // The behaviour of getOffsetFromLocal is defined in its javadoc
+    // However, this definition doesn't work for all DateTimeField operations
+    
+    /** Mock zone simulating Asia/Gaza cutover at midnight 2007-04-01 */
+    private static long CUTOVER_GAZA = 1175378400000L;
+    private static int OFFSET_GAZA = 7200000;  // +02:00
+    private static final DateTimeZone MOCK_GAZA = new MockZone(CUTOVER_GAZA, OFFSET_GAZA, 3600);
+
+    //-----------------------------------------------------------------------
+    public void test_MockGazaIsCorrect() {
+        DateTime pre = new DateTime(CUTOVER_GAZA - 1L, MOCK_GAZA);
+        assertEquals("2007-03-31T23:59:59.999+02:00", pre.toString());
+        DateTime at = new DateTime(CUTOVER_GAZA, MOCK_GAZA);
+        assertEquals("2007-04-01T01:00:00.000+03:00", at.toString());
+        DateTime post = new DateTime(CUTOVER_GAZA + 1L, MOCK_GAZA);
+        assertEquals("2007-04-01T01:00:00.001+03:00", post.toString());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 62:67, 271:276, 489:494, 636:641, 820:825, 881:886, 962:967.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.